pkg/apis/kops/validation/legacy.go

```diff
@@ -400,6 +400,34 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 		}
 	}
 
+	if c.Spec.IAMRolesForServiceAccounts != nil {
+		allErrs = append(allErrs, validateIRSA(c.Spec.IAMRolesForServiceAccounts, fieldSpec.Child("iamRolesForServiceAccounts"))...)
+	}
+	return allErrs
+}
+
+func validateIRSA(irsa *kops.IAMRolesForServiceAccountsConfig, spec *field.Path) (allErrs field.ErrorList) {
+	if !featureflag.PublicJWKS.Enabled() {
+		allErrs = append(allErrs, field.Forbidden(spec, "IAM roles for ServiceAccounts requires PublicJWKS feature flag to be enabled"))
+	}
+
+	sas := make(map[string]string)
+	for _, sa := range irsa.ServiceAccounts {
+		p := spec.Child("serviceAccounts")
+		key := fmt.Sprintf("%s/%s", sa.Namespace, sa.Name)
```

Comment:
Validate `Namespace` and `Name` are non-empty? Have `Namespace` default to `default`?